### DISCUSSION TITLE
Produktanlage PHP 8.x

### DISCUSSION THREAD

**Discussion by bigfraggle** [scheb/open-apexx#55] [General]
Da ja gerade mehrfach PHP8 Probleme angefragt werden, habe ich heute auch mal mein Hauptprojekt versucht zu aktualisieren.

Ein Fehler im Admin, der bislang noch nicht angesprochen wurde: Die Produktanlage ist nicht möglich. Wenn man auf "Produkt hinzufügen" klickt, passiert nichts. Außer Depricated-Hinweisen und Warnungen gibt es keine Infos, in der Konsole wird lediglich ein 500er in Bezug auf die admin/action.php angezeigt.

Produkte bearbeiten geht, News anlegen z. B. auch.

Wäre klasse, wenn jemand weiterhelfen kann.

**Comment by niclasd** [comment]
Moin @bigfraggle 
vorweg: Ich bin leider kein Programmierer und habe ja auch das [ein oder andere Problemchen](https://github.com/scheb/open-apexx/issues?q=is%3Aissue+is%3Aclosed+PHP+8) auf der Webseite meines Kumpels, der noch voll auf apexx setzt. Komischerweise hat er mir dieses Problem bislang nicht geschildert, sodass ich hier hellhörig wurde und mal nachgesehen habe:

Ich konnte den von dir gemeldeten Fehler reproduzieren und bekam eine strict-origin-when-cross-origin-Meldung in der Konsole angezeigt. Daraufhin habe ich versucht, das Problem über htaccess zu lösen - erfolglos.

Hab dann mal nachgefragt mit welchem Browser verwendet, um die Produkte einzustellen: Edge und Firefox -> hatte ich nach Chrome allerdings auch direkt probiert und hatte keinen Erfolg.

Schlussendlich hat sich herausgestellt, dass ich nach Auswahl einiger Felder unter "Optional" (Entwickler/Verlag bei Videospiel) auch das Produkt anlegen konnte. Bei einer anderen Kategorie ("Musik") bin ich damit aber nicht weitergekommen.

Außerdem sind mir in der Konsole zwei Hinweise zur Vermischung von http/https aufgefallen, die vielleicht Ursache sein könnten?`

Das erstmal zu meinen Erkenntnissen :)

`Gemischte (unsichere) Anzeige-Inhalte von "https://www.stylemotion.de/kunden/version_apexx.gif" werden auf einer sicheren Seite geladen
 [action.php]`
`Das Cookie "apx_open[dd_products]" verfügt über keinen gültigen Wert für das "SameSite"-Attribut. Bald werden Cookies ohne das "SameSite"-Attribut oder mit einem ungültigen Wert dafür als "Lax" behandelt. Dadurch wird das Cookie nicht länger an Kontexte gesendet, die zu einem Drittanbieter gehören. Falls Ihre Anwendung das Cookie in diesen Kontexten benötigt, fügen Sie bitte das Attribut "SameSite=None" zu ihm hinzu. Weitere Informationen zum "SameSite"-Attribut finden Sie unter https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite. [sortablenavi.js:178]`

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite

**Comment by bigfraggle** [comment]
Hey @niclasd, lieben Dank für deine ausführliche Rückmeldung. Dass mit ein paar optionalen Angaben die Produkte angelegt werden können, ist mir nicht aufgefallen. Ich werde die hoffentlich ruhigen Tage nutzen und mir das noch einmal anschauen. :)

Die Versionsgrafik habe ich schon vor einiger Zeit mal rausgenommen. Die Info braucht man ja eh nicht mehr.

Dass mit dem Cookie ist sicherlich ein hilfreicher Hinweis, das wird mir hier nicht angezeigt oder zumindest habe ich es nicht in der Konsole gesehen. Da Drittanbieter hier keine Rolle spielen, sollte das eigentlich nicht so wichtig sein, oder?

**Comment by bigfraggle** [comment]
Hallo @niclasd, ich habe mir das jetzt mal angeschaut. Die Produktanlage und (!) auch -bearbeitung hakt hier tatsächlich nur an der Eingabe für Medienformat. Dies ist jetzt tatsächlich gewissermaßen ein Pflichtfeld geworden... Daran kann man sich bestimmt gewöhnen, falls aber jemand eine Idee für einen Fix hat, dann immer gerne her damit.

Verwirrend ist, dass das Feld "Systeme" ja technisch eigentlich identisch ist, dort aber nach wie vor keine Eingabe gemacht werden muss.
